src/base/baseBrushTool.js

```diff
@@ -0,0 +1,367 @@
+import external from './../externalModules.js';
+import EVENTS from './../events.js';
+import baseTool from './../base/baseTool.js';
+// Utils
+import KeyboardController from '../fancy-tools/shared/KeyboardController.js';
+import isToolActive from '../fancy-tools/shared/isToolActive.js';
+import { getNewContext } from '../util/drawing.js';
+import { COLOR_MAP_ID } from '../stateManagement/brushToolColors.js';
+
+export default class extends baseTool {
+  constructor ({
+    name,
+    strategies,
+    defaultStrategy,
+    configuration,
+    supportedInteractionTypes
+  }) {
+    super({
+      name,
+      strategies,
+      defaultStrategy,
+      configuration,
+      supportedInteractionTypes
+    });
+
+    this.isBrushTool = true;
+    this.hasCursor = true;
```

Comment:
Do we want _all_ brush tools to have a cursor (and in all states for that brush)? Is there a way to change this? Would it be worth separating out the `render` so that tools with a `renderCursor` method trigger the redraw and handle the rendering of the cursor so we can keep the logic separate from the primary tool rendering?

Reply:
I can't see a brush tool that wouldn't require some form of cursor so that you can see what you intend to paint, but there's always an exception, which I'd be happy to hear about if you have any idea.

As for the latter part of your question, I do this with renderBrush, but you're right it should be renamed renderCursor for consistency.